macOS: "ld: library not found for -lasan"

when i run `make` on macOS, I see:

```
gcc -g -c GPMF_demo.c
gcc -g -c ../GPMF_parser.c
gcc -g -c GPMF_mp4reader.c
gcc -g -c GPMF_print.c
gcc -o gpmfdemo GPMF_demo.o GPMF_parser.o GPMF_mp4reader.o GPMF_print.o -lasan
ld: library not found for -lasan
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [gpmfdemo] Error 1
```

[This comment](https://github.com/kovidgoyal/kitty/issues/1162#issuecomment-440908886) implies replacing [`-lasan`](https://github.com/gopro/gpmf-parser/blob/eeef670dfdc51d4761cfe654cfbe212444cc9d19/demo/makefile#L2) with `-fsanitize=address` works on `clang`, and indeed [it seems to get the demo building for me locally](https://github.com/ryan-williams/gpmf-parser/commit/e459a498071ad796027a23c8cede5da01aecc1dd).

I guess ideally the `makefile` would look at what OS it's on, and use the appropriate parameter.

--- a/demo/makefile
+++ b/demo/makefile
@@ -1,5 +1,15 @@
+# Address-sanitizer library
+ASAN_FLAGS := -lasan
+ifneq ($(OS),Windows_NT)
+	UNAME_S := $(shell uname -s)
+	ifeq ($(UNAME_S),Darwin)
+		# macOS requires different a flag
+		ASAN_FLAGS := -fsanitize=address
+	endif
+endif
+
 gpmfdemo : GPMF_demo.o GPMF_parser.o GPMF_mp4reader.o GPMF_print.o
-		gcc -o gpmfdemo GPMF_demo.o GPMF_parser.o GPMF_mp4reader.o GPMF_print.o -lasan
+		gcc -o gpmfdemo GPMF_demo.o GPMF_parser.o GPMF_mp4reader.o GPMF_print.o $(ASAN_FLAGS)
 
 GPMF_demo.o : GPMF_demo.c ../GPMF_parser.h
 		gcc -g -c GPMF_demo.c

